### PR TITLE
Add test for empty slab creation

### PR DESCRIPTION
**Pull request by jnferner**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -58,6 +58,12 @@ pub macro slab {
 mod test {
     use super::slab;
 
+    #[test]
+    fn slab_macro_can_create_empty_slab() {
+        let (slab, []): (slab::Slab<i32>, [usize; 0]) = slab![];
```
Comment (bash) [inline src/lib.rs]: I believe the second type in the tuple could be omitted like so: (slab::Slab<i32>, _)

Comment (jnferner) [reply src/lib.rs]: That's how I first wrote it, however:
image

Comment (jnferner) [reply src/lib.rs]: The type of `keys` is `[usize; element_count]`, where `element_count` is determined by `count_tts!()`, which is not a `const fn` on stable.

Comment (bash) [reply src/lib.rs]: I shall accept my defeat 😉

Comment (jnferner) [reply src/lib.rs]: If activate the feature `const_slice_len`, the tests don't compile, because the feature is not active for the tests, for some reason
